src/utils/hooks.js

```diff
@@ -19,3 +24,92 @@ export function useStateWithStorage(storageKey, initialValue) {
 
 	return [value, setValue];
 }
+
+export function useUrgency(items) {
+	const [urgencyObject, setUrgecyObject] = useState({
+		overdue: [],
+		soon: [],
+		kindOfSoon: [],
+		notSoon: [],
+		inactive: [],
+	});
+
+	const getUrgency = (name) => {
+		const statusArray = Object.entries(urgencyObject).find(
+			([status, items]) => {
+				if (items.some((item) => item.name === name)) {
+					return status;
+				}
+			},
+		);
+		if (!statusArray) {
+			throw new Error(`Failed to get class name of ${name}`);
+		}
+		return statusArray[0];
+	};
```

Comment:
Could we create a `urgencyUtils.js` under the utils directory and move this urgency-related function to this new file?

Reply:
I would like to abstract this function, but it relies on the on the urgency object and I am not sure how to move it out. I will push some updates, let me know what do you think!